airbyte-integrations/connectors/source-file/source_file/source.py

```diff
@@ -84,8 +84,10 @@ def _validate_and_transform(config: Mapping[str, Any]):
         if "reader_options" in config:
             try:
                 config["reader_options"] = json.loads(config["reader_options"])
+                if not isinstance(config["reader_options"], dict):
+                    raise ValueError
```

Comment:
I think we need specialised error message with more accurate info.

Reply:
message updated